src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -804,8 +804,43 @@ protected bool TakeStartLine(SocketInput input)
                 QueryString = queryString;
                 HttpVersion = httpVersion;
 
-                bool caseMatches;
+                if (requestUrlPath.Length > 0 && requestUrlPath[0] != '/')
+                {
+                    int hostIndex;
+                    if (requestUrlPath.StartsWith("http://"))
```

Comment:
Specify the StringComparison to avoid a CurrentCulture comparision. Is requestUrlPath expected to be lower case?